hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/store/contracts/precompile/codec/CustomFeeTest.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.evm.store.contracts.precompile.codec;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.junit.jupiter.api.Test;
+
+class CustomFeeTest {
+
+    @SuppressWarnings({"java:S5845", "java:S3415"})
```

Comment:
Generally speaking we should not be suppressing sonar warnings inline like this.  Only if the detector is broken and then there should be a comment explaining the justification.